core/unit_test/TestTeam.hpp

```diff
@@ -118,6 +118,18 @@ struct TestTeamPolicy {
         TestTeamPolicy());
   }
 
+  static void test_constructors() {
+    constexpr const int smallest_work = 1;
+    Kokkos::TeamPolicy<ExecSpace, NoOpTag> none_auto(
+        smallest_work, smallest_work, smallest_work);
+    Kokkos::TeamPolicy<ExecSpace, NoOpTag> both_auto(
+        smallest_work, Kokkos::AUTO(), Kokkos::AUTO());
+    Kokkos::TeamPolicy<ExecSpace, NoOpTag> auto_vector(
+        smallest_work, smallest_work, Kokkos::AUTO());
+    Kokkos::TeamPolicy<ExecSpace, NoOpTag> auto_team(
+        smallest_work, Kokkos::AUTO(), smallest_work);
+  }
```

Comment:
This only tests that we don't segfault. Can we test on a `parallel_for` and a `parallel_reduce` that the values make sense?

Reply:
we probably want somehwere a tuning test which runs different combos and checks whether the tuner actually found a combo delivering similar time.

Reply:
That should be its own unit test executable and its own header file ...